### DISCUSSION TITLE
Tasmota for ESP32 with T-SIM7000G

### DISCUSSION THREAD

**Discussion by midou16m** [arendst/Tasmota#16381] [Ideas]
it's time to progress Tasmota with a SIM module (T-SIM7000G / ESP32). gps + 4g lte + wifi + bluetooth. who will participate?

For example
https://ae01.alicdn.com/kf/H8409429311e840eda823ad1ecc2fd33cU/LILYGO-TTGO-T-SIM7000G-SIM-Development-Board-ESP32-Wireless-Module-WiFi-Bluetooth-GPS-Antenna-Support-Expansion.jpg

**Comment by Jason2866** [comment]
I edited the headline and your post. We do not want to be device specific where is no need for.

**Comment by sfromis** [comment]
While I'm not really into those SIM boards, my general expectation is that you can communicate with them over UART serial, using `AT` commands, which can work nicely with coding in Berry available on Tasmota with ESP32, allowing you to make what you want with a reasonably simple programming language.

**Reply by barbudor** [reply]
Managing the SIMxxxx with AT commands at application level have some interrest to do SMS or some https requests.

It could be more insterresting to leverage the PPP layers in the Arduino/Espressif core to integrate the broadband network at low level with the same features as Wifi or Ethernet. That would allow transparent access to all network features including MQTT, WebQuery, SendMail, InflluxDB, Berry network API, ...
Of course there are some security concerns and it is likely the Web UI should be disabled, or at least binded only to Wifi/Ethernet.
Still the use-case is very limited considering a 4G Wifi dongle only cost 30€ on Ali and can host around 10 devices.
I see 2 use case that could have some value:
- In-car to monitor some information, eventually with a GPS and a CAN interface
- Remote sensor, in an (almost)- off-grid location

Nevertheless, the Arduino/Espressif core used by Tasmota doesn't include the PPP layer.
A few months ago @Jason2866 gave me some details on how to build a core with the features but I was diverted to other tasks.

**Comment by RaphaelSquair** [comment]
I would also love to have this application working, using the ppp layer would also open the possibility of using Lora, ZigBee, and Thread via other modules, it is not an easy task to build a new core that supports this feature but it would expand the possibilities of the tasmota project to another level.

There is still someone working on this feature?

**Reply by Jason2866** [reply]
Enabling the PPP part in IDF and building a framework is easy. BUT thats just the IDF API. It needs implementation in Tasmota. Thats work, nobody has done ;-) and there are no plans to do.

**Reply by sfromis** [reply]
Of course, contributions welcome :wink:

**Reply by pkkrusty** [reply]
@barbudor first response still holds true...you can get a 4G usb dongle for $15-$30 with no new integration required. I agree it would be neat to have a robust driver for the SIM modules, but the dongles are quite convenient.

**Comment by GEAG** [comment]
- PPP well integrated into the lwip stack and usable over diverse serial streams (UART, USB, etc.) might be quite useful.
- Is anybody working on attaching Wifi HaLow (IEEE 802.11ah) modules to Tasmota for the EU regulated countries?
- There are currently several on-going efforts in Switzerland to interface Newracom and Morse Micro modules (or the EU regulation domain) over SPI to an ESP32.

**Comment by Jason2866** [comment]
@GEAG none of the core Tasmota developers is working on this. It should not be complicated since Tasmota does use actual espressif Arduino core which supports PPP. "Just" needs integration in Tasmota
https://github.com/espressif/arduino-esp32/tree/master/libraries/PPP

**Reply by sfromis** [reply]
Somewhat related (not at the PPP level), this Berry project https://github.com/pkoevesdi/berryfona/ talks about "GSM library" and "For SIM800L, maybe others"
